service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -585,6 +586,33 @@ public void createBookLog(JWTData userData, int schoolId, UpsertBookLogRequest r
     log.store();
   }
 
+  @Override
+  public void adminUpdateBookLog(
+      JWTData userData, int schoolId, int bookId, UpsertBookLogRequest request) {
+    if (!userData.isAdmin()) {
+      throw new AdminOnlyRouteException();
+    }
+    SchoolsRecord school = this.queryForSchool(schoolId);
+    if (school == null) {
+      throw new SchoolDoesNotExistException(schoolId);
+    }
+
+    BookLogsRecord log = db.selectFrom(BOOK_LOGS).where(BOOK_LOGS.ID.eq(bookId)).fetchOne();
+
+    if (log == null) {
+      throw new BookLogDoesNotExistException(bookId);
+    }
+
+    Integer count = request.getCount();
+    Timestamp date = request.getDate();
+    String notes = request.getNotes();
+    log.setSchoolId(schoolId);
+    log.setCount(count);
+    log.setDate(date);
+    log.setNotes(notes);
+    log.store();
```

Comment:
❓ no `.refresh()`?

Reply:
No `refresh()` needed here since we're not using the updated record!

Reply:
ngl, wasn't sure what refresh even did